.github/workflows/compile-check.yaml

```diff
@@ -0,0 +1,27 @@
+name: Compile & Check 
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+  workflow_dispatch: # Manual trigger added
+  workflow_call: # Allow other Actions to call this workflow 
+  
+jobs:
+  Compile-and-Check:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+    - name: Compile winutil.ps1
+      shell: pwsh
+      run: |
+        Set-ExecutionPolicy Bypass -Scope Process -Force; ./Compile.ps1
+      continue-on-error: false # Directly fail the job on error, removing the need for a separate check
+    - name: Check Syntax of winutil.ps1 
+      shell: pwsh
+      run: |
+        Set-ExecutionPolicy Bypass -Scope Process -Force; Get-Command -Syntax .\winutil.ps1 
```

Comment:
Now I know this's something somewhat overlooked, as it's both a lot of work to do, and to maintain it... but we should try adding a `winutil.ps1 -RunTests` (will need to make a `Debug` flag for `Compile.ps1` Script, so we can remove said tests from final released build), or have a full testing suite, which's ran after Syntax checking of `winutil.ps1`.

Overall, I do like this Syntax Check of WinUtil & _finally_ removing `winutil.ps1` from tracking list (as it's not needed anymore), cool PR @Marterich 👍

Reply:
As a quick fix we could also add the Get-Command -Syntax trick to the compile script and creating a new argument. This could streamline the GitHub Action and also provide the sanity check for every local compile

Reply:
Good idea @Marterich
Will see what I can do about it, and make a PR to your fork.

Reply:
@Marterich made a new PR to your fork, [link to it](https://github.com/Marterich/winutil/pull/29).